src/storage-client/src/storage_collections.rs

```diff
@@ -396,6 +409,7 @@ where
         _now: NowFn,
         txns_metrics: Arc<TxnMetrics>,
         envd_epoch: NonZeroI64,
+        read_only: bool,
```

Comment:
This value is important enough that should probably be an enum

Reply:
All the other places also use a `read_only: bool` (ComputeController, StorageController, Controller), so I'd like to keep it consistent with that for now.